File: pushfile.py
#!/usr/bin/env python

import paramiko
from paramiko import AutoAddPolicy, SSHClient

port = 22
username = 'username'
pkey_file = '/home/username/.ssh/id_rsa'
hosts = 'hosts.txt'
files = 'files.txt'

key = paramiko.RSAKey.from_private_key_file(pkey_file)
paramiko.util.log_to_file('paramiko.log')

for host in open(hosts):
    host=host.rstrip()
    client = paramiko.Transport((host, port))
    client.connect(username=username, pkey=key)
    sftp = paramiko.SFTPClient.from_transport(client)
    for file in open(files):
      file=file.rstrip()
      print "Pushing",file,"to",host
      sftp.put(file, file)
    client.close()

File: pushfile.rb
#!/usr/local/bin/ruby

require 'rubygems'
require 'net/sftp'

username='username'
hosts = 'hosts.txt'
pushfiles = 'files.txt'

File.open(hosts).each_line{ |host|
    host = host.chomp
    sftp = Net::SFTP.start(host, username)
  File.open(pushfiles).each_line{ |pushfile|
    pushfile = pushfile.chomp
    puts "Pushing #{pushfile} to #{host}.\n"
    sftp.upload!(pushfile, pushfile)
  }
}
